cg/store/crud/read.py

```diff
@@ -1528,3 +1545,67 @@ def get_case_ids_for_samples(self, sample_ids: list[int]) -> list[str]:
         for sample_id in sample_ids:
             case_ids.extend(self.get_case_ids_with_sample(sample_id))
         return list(set(case_ids))
+
+    def _get_related_dna_samples_from_sample_within_collaborators(
+        self, sample_internal_id: str, collaborators: set[Customer]
+    ) -> list[Sample]:
+        """Returns a DNA sample with the same subject_id, tumour status and within the collaborators of an RNA sample."""
+        sample_application_version_query: Query = self._get_join_sample_application_version_query()
+        sample_application_version_query: Query = apply_application_filter(
+            applications=sample_application_version_query,
+            filter_functions=[ApplicationFilter.IS_DNA],
+        )
+
+        sample: Sample = self.get_sample_by_internal_id(internal_id=sample_internal_id)
```

Comment:
For clarity I would get the sample as the first line of this method

Reply:
Changed!